nav2_collision_monitor/test/collision_detector_node_test.cpp

```diff
@@ -0,0 +1,598 @@
+// Copyright (c) 2022 Samsung R&D Institute Russia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <math.h>
+#include <cmath>
+#include <chrono>
+#include <memory>
+#include <utility>
+#include <vector>
+#include <string>
+#include <limits>
+
+#include "rclcpp/rclcpp.hpp"
+#include "nav2_util/lifecycle_node.hpp"
+#include "sensor_msgs/msg/laser_scan.hpp"
+#include "sensor_msgs/msg/point_cloud2.hpp"
+#include "sensor_msgs/msg/range.hpp"
+#include "sensor_msgs/point_cloud2_iterator.hpp"
+#include "geometry_msgs/msg/twist.hpp"
+#include "geometry_msgs/msg/polygon_stamped.hpp"
+
+#include "tf2_ros/transform_broadcaster.h"
+
+#include "nav2_collision_monitor/types.hpp"
+#include "nav2_collision_monitor/collision_detector_node.hpp"
+
+using namespace std::chrono_literals;
+
+static constexpr double EPSILON = 1e-5;
+
+static const char BASE_FRAME_ID[]{"base_link"};
+static const char SOURCE_FRAME_ID[]{"base_source"};
+static const char ODOM_FRAME_ID[]{"odom"};
+static const char FOOTPRINT_TOPIC[]{"footprint"};
+static const char SCAN_NAME[]{"Scan"};
+static const char POINTCLOUD_NAME[]{"PointCloud"};
+static const char RANGE_NAME[]{"Range"};
+static const char STATE_TOPIC[]{"collision_detector_state"};
+static const int MAX_POINTS{1};
+static const double SIMULATION_TIME_STEP{0.01};
+static const double TRANSFORM_TOLERANCE{0.5};
+static const double SOURCE_TIMEOUT{5.0};
+
+enum PolygonType
+{
+  POLYGON_UNKNOWN = 0,
+  POLYGON = 1,
+  CIRCLE = 2
+};
+
+enum SourceType
+{
+  SOURCE_UNKNOWN = 0,
+  SCAN = 1,
+  POINTCLOUD = 2,
+  RANGE = 3
+};
+
+class CollisionDetectorWrapper : public nav2_collision_monitor::CollisionDetector
```

Comment:
Code coverage of the testcase is `92.8%` which is less than average `~95%` on CM:
![total_code_coverage](https://github.com/ros-planning/navigation2/assets/60094858/39280523-d51b-4b7d-82e6-78e6d88571e2)

There are some cases in `CD::configurePolygons()` and `CD::configureSources()` yet to be covered:
![configurePolygons_coverage](https://github.com/ros-planning/navigation2/assets/60094858/61bfd82d-248d-4c80-a613-db2c62174aa2)
![configureSources_coverage](https://github.com/ros-planning/navigation2/assets/60094858/ebcb9c62-ae0c-49df-9af5-bf9a4c19b86e)

These cases logic could be copied from corner cases at `collision_monitor_node_test.cpp` as well (e.g. by making circle safety shape with PCL2 and range sensors published; also action type rather than "none" also could be easily checked).